File: languages/java/src/main/java/com/bloock/sdk/entity/identity/NumberAttribute.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;

public class NumberAttribute extends Attribute<Long> {
    public NumberAttribute(String id, Long value) {
        super(id, value);
    }

    public static NumberAttribute fromProto(IdentityEntities.NumberAttribute res) {
        return new NumberAttribute(res.getId(), res.getValue());
    }

    public IdentityEntities.NumberAttribute toProto() {
        return IdentityEntities.NumberAttribute.newBuilder()
                .setId(this.id)
                .setValue(this.value)
                .build();
    }
}

File: languages/go/entity/identityV2/number_attribute.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type NumberAttribute struct {
	Attribute
}

func NewNumberAttribute(key string, value int64) NumberAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return NumberAttribute{
		Attribute: a,
	}
}

func NewNumberAttributeFromProto(s *proto.NumberAttribute) NumberAttribute {
	if s == nil {
		return NumberAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return NumberAttribute{
		Attribute: a,
	}
}

func (s NumberAttribute) ToProto() *proto.NumberAttributeV2 {
	return &proto.NumberAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(int64),
	}
}
